src/bundler.ts

```diff
@@ -118,62 +116,138 @@ class Bundler {
     }
     return false;
   }
 
-  hide(node) {
+  /**
+   * Creates a hidden container <div> to which inlined content will be
+   * appended.
+   * TODO(usergenic): Give this a more intention-revealing name.
+   */
+  getHiddenNode(): ASTNode {
     const hidden = dom5.constructors.element('div');
     dom5.setAttribute(hidden, 'hidden', '');
     dom5.setAttribute(hidden, 'by-vulcanize', '');
-    this.removeElementAndNewline(node, hidden);
-    dom5.append(hidden, node);
+    return hidden;
+  }
+
+  /**
+   * Inline external scripts <script src="*">
+   */
+  async inlineScript(documentUrl: string, externalScript: ASTNode):
+      Promise<void> {
+    const rawUrl: string = dom5.getAttribute(externalScript, 'src')!;
+    const resolved = url.resolve(documentUrl, rawUrl);
+    const backingScript: ScannedDocument =
+        await this.analyzer._scanResolved(resolved);
+    const scriptContent = backingScript.document.contents;
+    dom5.removeAttribute(externalScript, 'src');
+    dom5.setTextContent(externalScript, scriptContent);
   }
 
   /**
-   * Replace htmlImport
+   * Inline external stylesheets <link type="text/css" href="*">
    */
-  async inlineImport(
-      documentUrl: string, htmlImport: ASTNode, analyzer: Analyzer,
+  async inlineCss(documentUrl: string, externalStylesheet: ASTNode):
+      Promise<void> {
+    const rawUrl: string = dom5.getAttribute(externalStylesheet, 'href')!;
+    const resolved = url.resolve(documentUrl, rawUrl);
+    const backingStylesheet: ScannedDocument =
+        await this.analyzer._scanResolved(resolved);
+    const stylesheetContent = backingStylesheet.document.contents;
+    dom5.removeAttribute(externalStylesheet, 'href');
+    dom5.setTextContent(externalStylesheet, stylesheetContent);
+  }
+
+  /**
+   * Inline external HTML files <link type="import" href="*">
+   * TODO(usergenic): Refactor method to simplify and encapsulate case handling
+   *     for hidden div adjacency etc.
+   */
+  async inlineHtmlImport(
+      documentUrl: string, htmlImport: ASTNode,
       inlined: Set<string>): Promise<void> {
-    const rawUrl: string = dom5.getAttribute(htmlImport, 'href');
+    const rawUrl: string = dom5.getAttribute(htmlImport, 'href')!;
     const resolved = url.resolve(documentUrl, rawUrl);
     if (inlined.has(resolved)) {
       dom5.remove(htmlImport);
       return;
     }
     inlined.add(resolved);
     const backingDocument: ScannedDocument =
-        await analyzer._analyzeResolved(resolved);
+        await this.analyzer._scanResolved(resolved);
     const documentAst = dom5.parseFragment(backingDocument.document.contents);
     this.pathResolver.resolvePaths(documentAst, resolved, documentUrl);
-    const importParent = htmlImport.parentNode;
+    let importParent: ASTNode;
+    if (matchers.afterHiddenDiv(htmlImport)) {
+      importParent = dom5.nodeWalkPrior(htmlImport, matchers.hiddenDiv)!;
+      dom5.remove(htmlImport);
```

Comment:
TODO: remove the `remove()` call when https://github.com/PolymerLabs/dom5/issues/35 is fixed